### PR TITLE
[WIP] condp, if-not, when-not and not=

### PR DESCRIPTION
**Pull request by yurrriq**
Per lfex/clj#15, lfex/clj#10, lfex/clj#11, lfex/clj#12 and lfex/clj#19.

``` lfe
(condp {{(lambda (a b) ...) | #'f/2 | #'m:f/2}} x
  y z
  q {{(lambda (a) ...) | #'f/1 | #'m:f/1}}
  {{optional default}})
```

> I'm very open to alternate strategies/decisions here, but I wanted to get an initial version written.
> It would be great if `not=` weren't a macro, but this way it can be variadic like its Clojure analog.

### CODE REVIEW

**Comment by yurrriq** [comment]
Should we rename `not=` to `not==`?

**Comment by oubiwann** [comment]
There are some test errors, so I'm guessing you're still working on this?

File: include/clj.lfe [hunk 1]
```diff
@@ -571,6 +571,70 @@
...
+  otherwise if `test` evaluates to `true`, return `undefined`."
+  (`(,test . ,body)
+   `(if ,test 'undefined (progn ,@body))))
+
+(defmacro not=
+  "Same as `(not (== ...))`."
+  (`(,x . ())       'false)
+  (`(,x ,y . ,more) `(not (== ,x ,y ,@more))))
+
+
```
Comment (oubiwann) [inline include/clj.lfe:637]: This is good stuff, man -- I'm super-excited about it :-)

File: src/clj.lfe [hunk 2]
```diff
@@ -10,7 +11,8 @@
     (repeat 2)
     (seq 1) (seq 2) (seq 3)
     (split-at 2)
-    (take 2)))
+    (take 2))
+  (export-macro condp if-not when-not not=))
```
Comment (oubiwann) [inline src/clj.lfe:15]: Yeah, I think these are good candidates for macro exports and not includes.

**Comment by yurrriq** [comment]
The `export-macro` approach has some kinks. I got distracted with it last night (and also with writing property tests for `conj` which is still local). I'll likely look into it more tonight.

**Comment by yurrriq** [comment]
The test failure is due to duplicate definitions between `include/clj.lfe` and `src/clj.lfe`, so we still have to decide which definitions should feel like part of the language and which should be called with a `clj:` prefix.

**Comment by oubiwann** [comment]
No worries on any count. My intent was to make sure I wasn't implicitly blocking a merge in the event you had been expecting one :-)

**Comment by yurrriq** [comment]
Nope. I'll rename the PR then remove "[WIP]" when I think it's ready.

**Pull request closed without merging**